module/VuFind/src/VuFind/ILS/Connection.php

```diff
@@ -1341,6 +1344,92 @@ public function callIlsWithFailover($methodName, $params)
         );
     }
 
+    /**
+     * Get timed blocks for a method from driver configuration
+     *
+     * @param string $methodName Method to check
+     * @param array  $params     Array of passed parameters
+     *
+     * @return array Array with keys 'start', 'end', 'recurring'
+     *               or empty array if no blocks are found
+     */
+    public function getMethodTimedBlocks(string $methodName, array $params = []): array
+    {
+        $functionConfig = $this->checkCapability('getConfig', ['TimedBlocks', $params])
+            ? $this->getDriver()->getConfig('TimedBlocks', $params)
+            : [];
+
+        if (!isset($functionConfig[$methodName])) {
+            return [];
+        }
+        $blocks = [];
+        foreach ($functionConfig[$methodName] as $block) {
+            if (!str_contains($block, '/')) {
+                continue;
+            }
+            [$start, $end] = explode('/', $block, 2);
+            $isDate = preg_match('/^\d{4}-\d{2}-\d{2}/', $start ?: $end);
+
+            if ($isDate) {
+                $startDate = $start ? new \DateTime($start) : null;
+                $noEndHours = empty(explode(' ', $end, 2)[1]);
+                if ($end && $noEndHours) {
+                    $end .= ' 23:59:59';
+                }
+                $endDate = $end ? new \DateTime($end) : null;
+                $blocks[] = [
+                    'start' => $startDate,
+                    'end' => $endDate,
+                    'recurring' => false,
+                ];
+            } else {
+                $startTime = new \DateTime($start);
+                $endTime = new \DateTime($end);
```

Comment:
Shouldn't these check for empty $start or $end and create the objects only for non-empty? Otherwise the check right below doesn't make sense.

Reply:
Also needs tests to ensure that mistakes like "/10:00:00" and "10:00:00/" are handled properly.